salt/modules/rpm.py

```diff
@@ -168,14 +168,34 @@ def verify(*packages, **kwargs):
               'g': 'ghost',
               'l': 'license',
               'r': 'readme'}
+    options_map = {
+        'nocaps': '--nocaps',
+        'nodeps': '--nodeps',
+        'nodigest': '--nodigest',
+        'nofiledigest': '--nofiledigest',
+        'nofiles': '--nofiles',
+        'nogroup': '--nogroup',
+        'nolinkto': '--nolinkto',
+        'nomode': '--nomode',
+        'nomtime': '--nomtime',
+        'nordev': '--nordev',
+        'noscripts': '--noscripts',
+        'nosignature': '--nosignature',
+        'nosize': '--nosize',
+        'nouser': '--nouser'
+    }
     ret = {}
     ignore_types = kwargs.get('ignore_types', [])
+    verify_options = kwargs.get('verify_options', [])
+    cmd = ['rpm']
+    for option in verify_options:
+        cmd.append(options_map.get(option, ''))
```

Comment:
OK, I updated #33317 so that ``pkg.verify`` now raises an exception when the retcode of the ``rpm -V`` command is nonzero and there are no changes, and the state now catches this exception.